paf-mvp-core-js/src/validation/json-validator.ts

```diff
@@ -0,0 +1,203 @@
+import { Log } from '@core/log';
+import { default as AjvClass, ErrorObject } from 'ajv';
+import fs from 'fs';
+import path from 'path';
+
+export type Json = any;
+export type JsonSchemaType = string;
+
+/**
+ * Specify all the json schema types.
+ */
+export abstract class JsonSchemaTypes {
+  static createSeedRequest: JsonSchemaType = 'post-seed-request.json';
+}
+
+/**
+ * Validatation structure for a JSON
+ * against a JSON schema.
+ *
+ * If isValid = true,  the 'value' is available,
+ * else the instance contains 'errors'.
+ */
+export interface JsonValidation {
```

Comment:
[FAR] I liked your "Try" semantic in another PR. Maybe you could reuse it here?

Reply:
Well I prefer this one for now. The definition of Try in scala includes only one error whereas here we can have many errors here. Moreover, it may require some knowledge from FP and other languages to make the code instantly transparent.

In comparison, `JsonValidation` is extremely boring which is a good thing for a piece of code (I paraphrase R. Martin).

So unless the whole team want it or even more FP design patterns (like Reader for dependency injection, Writer for logs, Free Monad for defining the endpoints), we can do it later.